### PR TITLE
Re-enable XML as request and response types within the SwaggerUI

### PR DESCRIPTION
**Pull request by sweikenb**
This MR adds the `application/xml` option to the SwaggerUI for requests and responses.

### Description (*)

The Magento WebApi is capable of handling XML as exchange format beside JSON but this format option is not shown/available in the SwaggerUI for requests and responses (only `application/json` is shown).

### Manual testing scenarios (*)

1. Open the SwaggerUI in the browser `/swagger`
2. Select `application/xml` as format for the request parameters AND/OR
3. Select `application/xml` as format for the response data

Screenshot 2019-05-27 at 17 08 05

Screenshot 2019-05-27 at 17 08 13

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=23025) <br/>All committers have signed the CLA.

**Comment by XxXgeoXxX** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @XxXgeoXxX. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @XxXgeoXxX, here is your [new Magento instance](https://pr-23025.instances.magento-community.engineering).
Admin access: [https://pr-23025.instances.magento-community.engineering/admin](https://pr-23025.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by XxXgeoXxX** [comment]
Could you please check my review and fix the next points:
1. Fix static tests
2. Declare the new methods and constant as private as possible.
3. Add strict types declaration
declare(strict_types=1);

**Review by XxXgeoXxX — COMMENTED** [review]
Could you please check my review and fix the next points:
1. Fix static tests
2. Declare the new methods and constant as private as possible.
3. Add strict types declaration
declare(strict_types=1);

File: app/code/Magento/Webapi/Model/Rest/Swagger/Generator.php [hunk 1]
```diff
@@ -41,6 +41,11 @@ class Generator extends AbstractSchemaGenerator
     /** Array signifier */
     const ARRAY_SIGNIFIER = '[0]';
 
+    /**
+     * Wrapper node for XML requests
+     */
+    const XML_SCHEMA_PARAMWRAPPER = 'request';
```
Comment (XxXgeoXxX) [inline app/code/Magento/Webapi/Model/Rest/Swagger/Generator.php]: use private const

File: app/code/Magento/Webapi/Model/Rest/Swagger/Generator.php [hunk 2]
```diff
@@ -193,6 +198,28 @@ protected function getGeneralInfo()
         ];
     }
 
+    /**
+     * @return string[]
+     */
+    protected function getConsumableDatatypes()
```
Comment (XxXgeoXxX) [inline app/code/Magento/Webapi/Model/Rest/Swagger/Generator.php]: Declare the new method as private.

File: app/code/Magento/Webapi/Model/Rest/Swagger/Generator.php [hunk 3]
```diff
@@ -193,6 +198,28 @@ protected function getGeneralInfo()
...
+        return [
+            'application/json',
+            'application/xml',
+        ];
+    }
+
+    /**
+     * @return string[]
+     */
+    protected function getProducibleDatatypes()
```
Comment (XxXgeoXxX) [inline app/code/Magento/Webapi/Model/Rest/Swagger/Generator.php]: Declare the new method as private.

**Comment by XxXgeoXxX** [comment]
@sweikenb
@gelanivishal 
Could you recheck my remarks and provide changes?

**Comment by sweikenb** [comment]
@XxXgeoXxX Thanks for the feedback! I will look into it and provide an update a.s.a.p.!

**Review by XxXgeoXxX — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @XxXgeoXxX, thank you for the review.
[ENGCOM-5192](https://jira.corp.magento.com/browse/ENGCOM-5192) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@sweikenb thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by soleksii** [comment]
Hi @sweikenb !

Could you please correct the misspelled words in the sentence (line 880) ?

"**_sure_**" instead "**_shure_**",
"**_format_**" instead "**_fromat_**"

Thanks!

**Pull request merged**